src/Admin/Controller/DischargeController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Forumify\PerscomPlugin\Admin\Controller;
+
+use Forumify\Core\Repository\UserRepository;
+use Forumify\PerscomPlugin\Admin\Form\DischargeType;
+use Forumify\PerscomPlugin\Perscom\Entity\Record\AssignmentRecord;
+use Forumify\PerscomPlugin\Perscom\Entity\Record\RankRecord;
+use Forumify\PerscomPlugin\Perscom\Entity\Record\ServiceRecord;
+use Forumify\PerscomPlugin\Perscom\Repository\AssignmentRecordRepository;
+use Forumify\PerscomPlugin\Perscom\Repository\PerscomUserRepository;
+use Forumify\PerscomPlugin\Perscom\Repository\RankRecordRepository;
+use Forumify\PerscomPlugin\Perscom\Repository\ServiceRecordRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+class DischargeController extends AbstractController
+{
+    public function __construct(
+        private readonly PerscomUserRepository $perscomUserRepository,
+        private readonly AssignmentRecordRepository $assignmentRecordRepository,
+        private readonly ServiceRecordRepository $serviceRecordRepository,
+        private readonly UserRepository $userRepository,
+        private readonly RankRecordRepository $rankRecordRepository,
+    ) {
+    }
+    
+    #[Route('/users/{identifier}/discharge', 'user_discharge')]
+    public function __invoke(Request $request, int $identifier): Response
+    {
+        $this->denyAccessUnlessGranted('perscom-io.admin.user.discharge');
+
+        $user = $this->userRepository->find($identifier);
+
+        $perscomUser = $this->perscomUserRepository->findOneBy(['user' => $user]);
```

Comment:
- Pretty sure the user in quick_actions is already the perscom user?
- Since the permission is a fixed string, we can use attributes
- If we use `id` as the parameter, we can let Symfony's argument resolver take care of fetching it and returning not found (also need to change `identifier` to `id` in the `quick_actions.html.twig` for this)

```suggestion
    #[Route('/users/{id}/discharge', 'user_discharge')]
    #[IsGranted('perscom-io.admin.user.discharge')]
    public function __invoke(Request $request, PerscomUser $perscomUser): Response
    {
```